distributed/protocol/compression.py

```diff
@@ -124,14 +124,22 @@ def byte_sample(b, size, n):
     n : int
         number of samples to collect
     """
+    if n == 0:
+        return b""
+
+    b = ensure_memoryview(b)
+
     starts = [random.randint(0, len(b) - size) for j in range(n)]
     ends = []
     for i, start in enumerate(starts[:-1]):
         ends.append(min(start + size, starts[i + 1]))
     ends.append(starts[-1] + size)
 
     parts = [b[start:end] for start, end in zip(starts, ends)]
-    return b"".join(map(ensure_bytes, parts))
+    if n == 1:
+        return parts[0]
```

Comment:
Currently this returns a `memoryview`. This should work fine in `maybe_compress` below. If for some reason we want this to be `bytes`, we could copy, but that shouldn't be needed.

Reply:
I was assuming the copy was not wanted in my previous comment (I'm pretty sure `b"".join()` and `bytes()` are pretty equivalent).

Reply:
Yep agreed. Would prefer to avoid the copy and think we can without issues.